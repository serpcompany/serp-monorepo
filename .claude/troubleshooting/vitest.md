https://vitest.dev/guide/common-errors.html