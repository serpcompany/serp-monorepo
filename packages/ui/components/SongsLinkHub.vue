<template>
  <div class="mt-2 grid grid-cols-1 gap-2 md:grid-cols-2 xl:grid-cols-2">
    <div v-for="song in songs" :key="song.slug">
      <nuxt-link :to="`/songs/${song.slug}/`" aria-label="song name">
        {{ song.name }} -
        <span v-for="artist in song.artists" :key="artist.slug">
          {{ artist.credit_name }}
          <span v-if="artist.join_phrase">{{ artist.join_phrase }}</span>
        </span>
      </nuxt-link>
    </div>
  </div>
</template>

<script setup lang="ts">
import type { RecordingIndex } from '@serp/types/types';

defineProps({
  songs: {
    type: Array as () => RecordingIndex[],
    required: true
  }
});
</script>
