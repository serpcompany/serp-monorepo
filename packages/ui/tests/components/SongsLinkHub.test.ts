import { describe, expect, it } from 'vitest'
import SongsLinkHub from '../../components/SongsLinkHub.vue'
import ComponentRender from '../componentRender'

describe('songsLinkHub', () => {
  it.each([
    // No songs provided
    ['with no songs', { props: { songs: [] } }],
    // Single song with a single artist (no join phrase)
    [
      'with one song with one artist',
      {
        props: {
          songs: [
            {
              slug: 'song-1',
              name: 'Song One',
              artists: [
                {
                  slug: 'artist-1',
                  credit_name: 'Artist One',
                  join_phrase: '',
                },
              ],
            },
          ],
        },
      },
    ],
    // Multiple songs with multiple artists and join phrases
    [
      'with multiple songs and multiple artists',
      {
        props: {
          songs: [
            {
              slug: 'song-1',
              name: 'Song One',
              artists: [
                {
                  slug: 'artist-1',
                  credit_name: 'Artist One',
                  join_phrase: ', ',
                },
                {
                  slug: 'artist-2',
                  credit_name: 'Artist Two',
                  join_phrase: '',
                },
              ],
            },
            {
              slug: 'song-2',
              name: 'Song Two',
              artists: [
                {
                  slug: 'artist-3',
                  credit_name: 'Artist Three',
                  join_phrase: ' feat. ',
                },
                {
                  slug: 'artist-4',
                  credit_name: 'Artist Four',
                  join_phrase: '',
                },
              ],
            },
          ],
        },
      },
    ],
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `SongsLinkHub ${desc}`,
        options,
        SongsLinkHub,
      )
      expect(html).toMatchSnapshot()
    },
  )
})
