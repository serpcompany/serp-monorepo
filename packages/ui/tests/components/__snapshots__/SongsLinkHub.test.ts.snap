// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongsLinkHub > renders with multiple songs and multiple artists correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
">
  <div>
    <a
      aria-label="song name"
      href="/songs/song-1/"
    ><span>Song One - </span></a>
  </div>
  <div>
    <a
      aria-label="song name"
      href="/songs/song-2/"
    ><span>Song Two - </span></a>
  </div>
</div>
`;

exports[`SongsLinkHub > renders with no songs correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
"></div>
`;

exports[`SongsLinkHub > renders with one song with one artist correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  md:grid-cols-2
  mt-2
  xl:grid-cols-2
">
  <div>
    <a
      aria-label="song name"
      href="/songs/song-1/"
    ><span>Song One - </span></a>
  </div>
</div>
`;
